### ISSUE TITLE
Lumberjack Input not handling tags-parameter

### ISSUE THREAD

**Issue by dploeger** [elasticsearch/logstash#2004]
Either I completely misunderstood the lumberjack input, or it doesn't add the tags I specify in the "tags"-parameter.

My config:

```
input {

        lumberjack {

                port => 5000
                tags => ["lumberjack"]
                ssl_certificate => "/etc/ssl/certs/mycertificate.crt"
                ssl_key => "/etc/ssl/private/mykey.pem"

        }

}
```

I would expect, that each event reaching Logstash through this lumberjack input should be tagged with "lumberjack". In fact, it isn't. It only uses the tags supplied on logstash-forwarder on the other side.

**Comment by ph** [comment]
The current plugin logics concerning the tags is to always merge them with the existing one.
You are sure you are not using the field `type` on the forwarder side?, the `type` field will never be replaced by any input plugin.

Could you paste your `logstash-forwarder` configuration?

**Comment by dploeger** [comment]
Yes, I used the tags-field.

My logstash-forwarder-config:

```
{
  "network": {
    "servers": [ "hosea.office.getit.de:5000" ],
    "timeout": 15
  },
  "files": [
    {
      "paths": [ "/var/log/syslog"],
      "fields": {
	  "type": "test",
          "tags": "syslog"
      }
    }
  ]
}
```

My lumberjack input-configuration:

```
lumberjack {

        port => 5000
        tags => ["lumberjack"]
        ssl_certificate => "certifcate.crt"
        ssl_key => "key.pem"

    }
```

I would expect, that every log item, that reaches the server using lumberjack would get two tags: "syslog" (from the logstash-forwarder.conf) and "lumberjack" (from the input-configuration).

Actually, only the "syslog"-tag is carried through.

**Comment by ph** [comment]
https://github.com/logstash-plugins/logstash-input-lumberjack/blob/master/lib/logstash/inputs/lumberjack.rb#L49

**Comment by ph** [comment]
I can explain the behavior:

1. the event is received in the `lumberjack` input
2. we decode the communication
3. we decorate the event, the decoration process is where we should merge the tags.
4. we take the fields from the decoded event and override the field in the event. So we are overriding the `tags` field.

step 4 should be done before step 3. See https://github.com/logstash-plugins/logstash-input-lumberjack/blob/master/lib/logstash/inputs/lumberjack.rb#L49

Would you mind creating a new issue in the plugin repository at https://github.com/logstash-plugins/logstash-input-lumberjack ?

**Comment by dploeger** [comment]
No problem. Created logstash-plugins/logstash-input-lumberjack#6
